cmake/OmimSubdirectory.cmake

```diff
@@ -0,0 +1,71 @@
+set_property(GLOBAL PROPERTY USE_FOLDERS TRUE)
+
+# omim_add_subdirectory(subdir_name subdir)
+# add_subdirectory(subdir) and put all targets under the "subdir_name" folder
+function(omim_add_subdirectory)
+  # Documentation
+  # This function extends the functionality of add_subdirectory by grouping targets.
+  # If two arguments are provided, the first argument is considered the subdir_name.
+  # The second argument is the actual subdirectory name where add_subdirectory is called.
+  # If only one argument is provided, both subdir_name and subdir will take the same value.
+
+  if(ARGC EQUAL 2)
+    set(subdir_name ${ARGV0})
+    set(subdir "${ARGV1}")
+  elseif(ARGC EQUAL 1)
+    set(subdir_name ${ARGV0})
+    set(subdir ${ARGV0})
+  else()
+    message("Invalid number of arguments.")
+  endif()
+
+  add_subdirectory(${subdir})
+  _omim_get_all_targets_in_dir(_targets "${subdir}")
+  foreach(_target IN LISTS _targets)
```

Comment:
This looks a bit hacky to achieve visual changes in IDE... What doesn't work if we keep original add_subdirectory() everywhere?

Reply:
Think of it as a generic solution to group targets recursively. It works even if the project in the subdirectory doesn't support this feature by default.

> What doesn't work if we keep original add_subdirectory() everywhere?

Everything will work except that we'll have a mess in the targets list as shown in the "Before" screenshot.
Some libs (especially 3party) produce a lot of targets that may be impossible to determine what library/module they belong to.

F.e.  `gflags`. That's what it looks like after my changes.

<img width="240" alt="Screenshot 2023-05-19 at 12 19 02" src="https://github.com/organicmaps/organicmaps/assets/10351358/f7c57add-697c-4c37-b1c2-602cff47d825">


Another example is `protobuf`. We have now a very old version of it (v3, I guess). I'd like to update it to the new one (v23) and link it as a submodule directly. It should fix build warnings when we'll migrate to C++20. But it will produce dozen of targets (with meaningful names) that will mess up our targets list even more. So, that will be a good solution to  store all related targets under the `3party/protobuf/...` folder.